src/NServiceBus.Core/Hosting/Helpers/AssemblyScanner.cs

```diff
@@ -81,11 +83,18 @@ public AssemblyScannerResults GetScannableAssemblies()
 
             var assemblies = new List<Assembly>();
 
-            foreach (var assemblyFile in ScanDirectoryForAssemblyFiles(baseDirectoryToScan, ScanNestedDirectories))
+            var directoriesToScan = string.IsNullOrWhiteSpace(AdditionalAssemblyScanningPath)
```

Comment:
While this code path is only called once on startup, I'm wondering if it would make sense to refactor the code below into a method that can be called instead of allocating an array just to use a `foreach`.

Reply:
I can update that. See how it looks like in the next commit.